internal/femto/authentication.go

```diff
@@ -0,0 +1,56 @@
+package femto
+
+import (
+	"errors"
+	"log/slog"
+	"net/http"
+)
+
+var (
+	NotAuthenticatedError   = errors.New("User does not have a valid authentication token")
+	InvalidCredientalsError = errors.New("Invalid credientals for creating an authentication token")
+)
+
+const (
+	AuthCookieField = "auth_cookie"
+)
+
+type AuthToken = string
+
+type Authenticator[T any] interface {
+	CreateToken(T) (AuthToken, error)
+	RevokeToken(token AuthToken) error
+	CheckToken(token AuthToken) bool
+}
+
+func AuthWrapGet[A any, T any](auth Authenticator[A], handle GetFunc[T]) GetFunc[T] {
+	return func(r *http.Request, l *slog.Logger) (T, error) {
+		var zero T
+		cookie, err := r.Cookie(AuthCookieField)
+		if err != nil {
+			return zero, err
```

Comment:
Could you wrap this error to document the cookie you expected? Go's [ErrNoCookie](https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/net/http/request.go;l=420) doesn't have this detail